frontend/src/lib/SegmentMetricsManager.ts

```diff
@@ -211,7 +212,23 @@ export class SegmentMetricsManager implements MetricsManager {
 
   // eslint-disable-next-line class-methods-use-this
   private track(evName: string, data: Record<string, unknown>): void {
-    analytics.track(evName, data)
+    let parser = new UAParser("user-agent")
+    let parsed_ua = parser.getResult()
+
+    analytics.track(evName, data, {
+      context: {
+        ip: "0.0.0.0",
+        page: {
+          title: "",
+        },
+        browser: parsed_ua.browser.name,
```

Comment:
nit: UAParser also has the `engine` information. I believe for development purposes this is actually a bit more useful than tracking the browser. So maybe we also add the engine & engine version?

Reply:
Added engine